cartographer/internal/mapping/global_trajectory_builder.h

```diff
@@ -59,17 +59,26 @@ class GlobalTrajectoryBuilder : public mapping::TrajectoryBuilderInterface {
       return;
     }
     std::unique_ptr<mapping::NodeId> node_id;
+    std::unique_ptr<InsertionResult> insertion_result;
     if (matching_result->insertion_result != nullptr) {
       node_id = ::cartographer::common::make_unique<mapping::NodeId>(
           pose_graph_->AddNode(
               matching_result->insertion_result->constant_data, trajectory_id_,
               matching_result->insertion_result->insertion_submaps));
       CHECK_EQ(node_id->trajectory_id, trajectory_id_);
+      insertion_result = common::make_unique<InsertionResult>();
+      insertion_result->constant_data =
+          matching_result->insertion_result->constant_data;
+      for (const auto& insertion_submap :
```

Comment:
Why not just
```
insertion_result->insertion_submaps = matching_result->insertion_result->insertion_submaps
```
or
```
common::make_unique<InsertionResult>(InsertionResult{
  matching_result->insertion_result->constant_data;
  matching_result->insertion_result->insertion_submaps;
});
```

Reply:
That does not compile.

Note that the `matching_result->insertion_result->insertion_submaps` are of type `std::vector<mapping_2d/3d::Submap>` while the newly introduced one's are `std::vector<mapping::Submap>`, i.e. the base class.

Reply:
I see. What about the range constructor then, i.e.
```
common::make_unique<InsertionResult>(InsertionResult{
  matching_result->insertion_result->constant_data,
  std::vector<std::shared_ptr<const Submap>>(
    matching_result->insertion_result->insertion_submaps.begin(),
    matching_result->insertion_result->insertion_submaps.end())});
```
?